handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -617,6 +597,92 @@ public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws E
         ctx.fireExceptionCaught(cause);
     }
 
+    /**
+     * Logger for TCP
+     */
+    private void logTCP(boolean isWriteOperation, int bytes, int sendSegmentNumber, int receiveSegmentNumber,
+                        InetSocketAddress srcAddr, InetSocketAddress dstAddr, boolean ackOnly) {
+        // If `ackOnly` is `true` when we don't need to write any data so we'll not
+        // log number of bytes being written and mark the operation as "TCP ACK".
+        if (logger.isDebugEnabled()) {
+            if (ackOnly) {
+                logger.debug("Writing TCP ACK, isWriteOperation {}, Segment Number {}, Ack Number {}, Src Addr {}, "
+                        + "Dst Addr {}", isWriteOperation, sendSegmentNumber, receiveSegmentNumber, dstAddr, srcAddr);
+            } else {
+                logger.debug("Writing TCP Data of {} Bytes, isWriteOperation {}, Segment Number {}, Ack Number {}, " +
+                                "Src Addr {}, Dst Addr {}", bytes, isWriteOperation, sendSegmentNumber,
+                        receiveSegmentNumber, srcAddr, dstAddr);
+            }
+        }
+    }
+
+    OutputStream outputStream() {
+        return outputStream;
+    }
+
+    boolean sharedOutputStream() {
+        return sharedOutputStream;
+    }
+
+    public State state() {
```

Comment:
```suggestion
    State state() {
```

Reply:
If we make this package-private then people won't be able to access JavaDoc returned by `state` method?

Reply:
the state method should also be `package` private ?

Reply:
imho the user should just need `isPaused()` or something like that... no need to expose the whole state

Reply:
Alright, `isPaused()` sounds better.